File: iOS OC Example/iOS OC Example/NativeViewController.m
//
//  NativeViewController.m
//  iOS OC Example
//
//  Created by M on 2019/5/28.
//  Copyright © 2019 1tu1shu. All rights reserved.
//

#import "NativeViewController.h"
#import <AdTimingSDK/AdTimingSDK.h>

@interface NativeViewController ()<AdTimingNativeDelegate>

@property (nonatomic, strong) AdTimingNative *native;
@property (nonatomic, strong) AdTimingNativeAd *nativeAd;
@property (nonatomic, strong) AdTimingNativeView *nativeView;
@property (nonatomic, strong) UIImageView *iconView;
@property (nonatomic, strong) UILabel *titleLabel;
@property (nonatomic, strong) UILabel *bodyLabel;

@property(nonatomic,strong)UIButton *loadBtn;
@property(nonatomic,strong)UIButton *showBtn;
@property(nonatomic,strong)UILabel *logLabel;

@end

@implementation NativeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Native";
    self.view.backgroundColor = [UIColor whiteColor];
    
    self.loadBtn = [UIButton buttonWithType:UIButtonTypeSystem];
    self.loadBtn.frame = CGRectMake(20, 120, 120, 30);
    [self.loadBtn setTitle:@"Load" forState:UIControlStateNormal];
    self.loadBtn.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
    self.loadBtn.backgroundColor = [UIColor whiteColor];
    self.loadBtn.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    self.loadBtn.layer.cornerRadius = 5;
    [self.loadBtn addTarget:self action:@selector(loadBtnDidClicked) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:self.loadBtn];
    
    self.showBtn = [UIButton buttonWithType:UIButtonTypeSystem];
    self.showBtn.frame = CGRectMake(20, CGRectGetMaxY(self.loadBtn.frame)+20, 120, 30);
    [self.showBtn setTitle:@"Show" forState:UIControlStateNormal];
    self.showBtn.contentHorizontalAlignment = UIControlContentHorizontalAlignmentLeft;
    self.showBtn.backgroundColor = [UIColor whiteColor];
    self.showBtn.titleLabel.font = [UIFont boldSystemFontOfSize:18];
    self.showBtn.layer.cornerRadius = 5;
    [self.showBtn addTarget:self action:@selector(showBtnDidClicked) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:self.showBtn];
    
    self.logLabel = [[UILabel alloc] initWithFrame:CGRectMake(20, CGRectGetMaxY(self.showBtn.frame)+20, self.view.frame.size.width-40, 30)];
    self.logLabel.text = @"Log";
    self.logLabel.font = [UIFont systemFontOfSize:15];
    [self.view addSubview:self.logLabel];
    
    // Create ADTNative
    self.native = [[AdTimingNative alloc]initWithPlacementID:@"109"];
    self.native.delegate = self;
    [self.view addSubview:self.nativeView];
    
}

- (AdTimingNativeView*)nativeView{
    if(!_nativeView){
        _nativeView = [[AdTimingNativeView alloc]initWithFrame:CGRectMake(0,300, self.view.frame.size.width, 300)];
        _nativeView.mediaView = [[AdTimingNativeMediaView alloc]initWithFrame:CGRectMake(0, 0, self.view.bounds.size.width, 300)];
        [_nativeView addSubview:_nativeView.mediaView];
        _iconView = [[UIImageView alloc]initWithFrame:CGRectMake(0, 0, 40, 40)];
        [_nativeView addSubview:_iconView];
        _titleLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, 270, self.view.frame.size.width, 15)];
        _titleLabel.font = [UIFont systemFontOfSize:13];
        [_nativeView addSubview:_titleLabel];
        _bodyLabel = [[UILabel alloc]initWithFrame:CGRectMake(0, 285, self.view.frame.size.width, 15)];
        _bodyLabel.font = [UIFont systemFontOfSize:13];
        [_nativeView addSubview:_bodyLabel];
    }
    return _nativeView;
}


-(void)loadBtnDidClicked
{
    // loadNative
    [self.native loadAd];
}

-(void)showBtnDidClicked
{
    // showNative
    self.nativeView.hidden = NO;
    self.nativeView.nativeAd = self.nativeAd;
    _iconView.image =[UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:self.nativeAd.iconUrl]]];
    _titleLabel.text = self.nativeAd.title;
    _bodyLabel.text = self.nativeAd.body;
}

/// Invoked when the ad is available.
/// You can then show the ad.
- (void)adtimingNative:(AdTimingNative*)native didLoad:(AdTimingNativeAd*)nativeAd{
    NSLog(@"NativeAd Did Load");
    self.nativeAd = nativeAd;
    self.logLabel.text = @"load success";
}

/// Invoked when the call to load an ad has failed.
/// Parameter error contains the reason for the failure.
- (void)adtimingNative:(AdTimingNative*)native didFailWithError:(NSError*)error{
    NSLog(@"NativeAd Did Fail");
}

/// Invoked when the Ad begins to show.
- (void)adtimingNativeWillExposure:(AdTimingNative*)native{
    NSLog(@"NativeAd Will Exposure");
}

/// Invoked when the ad finishes playing.
- (void)adtimingNativeDidClick:(AdTimingNative*)native{
    NSLog(@"NativeAd Did Click");
}


@end

File: iOS OC Example/iOS Swift Example/NativeViewController.swift
//
//  NativeViewController.swift
//  iOS Swift Example
//
//  Created by M on 2019/5/28.
//  Copyright © 2019 1tu1shu. All rights reserved.
//

import UIKit
import AdTimingSDK

class NativeViewController: UIViewController, AdTimingNativeDelegate {

    private lazy var iconView: UIImageView = {
        let icon = UIImageView(frame: CGRect(x: 0, y: 0, width: 40, height: 40))
        return icon
    }()
    
    private lazy var titleLabel: UILabel = {
        let titleLabel = UILabel(frame: CGRect(x: 0, y: 270, width: UIScreen.main.bounds.size.width, height: 15))
        titleLabel.font = UIFont.systemFont(ofSize: 13)
        return titleLabel
    }()
    
    private lazy var bodyLabel: UILabel = {
        let bodyLabel = UILabel(frame: CGRect(x: 0, y: 285, width: UIScreen.main.bounds.size.width, height: 15))
        bodyLabel.font = UIFont.systemFont(ofSize: 13)
        return bodyLabel
    }()
    
    var nativeAd: AdTimingNativeAd?
    
    private lazy var adtNativeView: AdTimingNativeView = {
        let view = AdTimingNativeView(frame: CGRect(x: 0, y: 300, width: self.view.frame.size.width, height: 300))
        view.mediaView = AdTimingNativeMediaView(frame: CGRect(x: 0, y: 0, width: self.view.frame.size.width, height: 300))
        view.addSubview(view.mediaView)
        view.addSubview(self.iconView)
        view.addSubview(self.titleLabel)
        view.addSubview(self.bodyLabel)
        
        return view
    }()
    
    private lazy var native: AdTimingNative = {
        let native = AdTimingNative(placementID: "109")
        native.delegate = self
        self.view.addSubview(adtNativeView)
        return native
    }()
    
    var logLabel = UILabel()
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.title = "Native"
        self.view.backgroundColor = UIColor.white
        
        let loadBtn:UIButton = UIButton(frame: CGRect(x: 20, y: 120, width: 120, height: 30))
        loadBtn.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.left
        loadBtn.backgroundColor = UIColor.white
        loadBtn.setTitleColor(UIColor.blue, for: .normal)
        loadBtn.setTitleColor(UIColor.gray, for: .highlighted)
        loadBtn.setTitleColor(UIColor.gray, for: .selected)
        loadBtn.titleLabel?.font = UIFont.systemFont(ofSize: 17)
        loadBtn.setTitle("Load", for: .normal)
        loadBtn.addTarget(self, action: #selector(loadBtnDidClicked), for: .touchUpInside)
        self.view.addSubview(loadBtn)
        
        let showBtn:UIButton = UIButton(frame: CGRect(x: 20, y: loadBtn.frame.maxY+20, width: 120, height: 30))
        showBtn.contentHorizontalAlignment = UIControl.ContentHorizontalAlignment.left
        showBtn.backgroundColor = UIColor.white
        showBtn.setTitleColor(UIColor.blue, for: .normal)
        showBtn.setTitleColor(UIColor.gray, for: .highlighted)
        showBtn.setTitleColor(UIColor.gray, for: .selected)
        showBtn.titleLabel?.font = UIFont.systemFont(ofSize: 17)
        showBtn.setTitle("Show", for: .normal)
        showBtn.addTarget(self, action: #selector(showBtnDidClicked), for: .touchUpInside)
        self.view.addSubview(showBtn)
        
        logLabel = UILabel(frame: CGRect(x: 20, y: showBtn.frame.maxY+20, width: 120, height: 30))
        logLabel.text = "Log"
        logLabel.font = UIFont.systemFont(ofSize: 15)
        self.view.addSubview(logLabel)
    }

    @objc func loadBtnDidClicked(){
        self.native.loadAd()
    }
    
    @objc func showBtnDidClicked(){
        self.adtNativeView.isHidden = false
        self.adtNativeView.nativeAd = self.nativeAd!
        self.iconView.image = UIImage(data: try! Data(contentsOf: URL(string: self.nativeAd?.iconUrl ?? "")!))
        self.titleLabel.text = self.nativeAd?.title
        self.bodyLabel.text = self.nativeAd?.body
    }
    
    func adtimingNative(_ native: AdTimingNative, didLoad nativeAd: AdTimingNativeAd) {
        print("nativeAd load success")
        self.nativeAd = nativeAd;
        self.logLabel.text = "load success"
    }
    
    func adtimingNative(_ native: AdTimingNative, didFailWithError error: Error) {
        print("nativeAddidFail")
        self.logLabel.text = "load fail"
    }
    
    func adtimingNativeWillExposure(_ native: AdTimingNative) {
        print("nativeAdWillExposure")
        self.logLabel.text = ""
    }
    
    func adtimingNativeDidClick(_ native: AdTimingNative) {
        print("nativeAdDidClick")
    }
    

}


